unittests/test_schedulers.py

```diff
@@ -166,6 +166,11 @@ def test_poll_before_submit(self):
         self.testjob.prepare(self.commands, self.environs)
         self.assertRaises(JobNotStartedError, self.testjob.finished)
 
+    def test_no_empty_lines_in_preamble(self):
+        self.setup_job()
```

Comment:
You should not call `setup_job()` here, because this sets everything and most probably the following test will have no meaning. I suggest omitting completely this call. In this case, `self.testjob` will be minimal.